src/MinimalDelegation.sol

```diff
@@ -2,5 +2,14 @@
 pragma solidity ^0.8.23;
 
 import {IMinimalDelegation} from "./interfaces/IMinimalDelegation.sol";
+import {Key, KeyLib} from "./lib/KeyLib.sol";
+import {MinimalDelegationStorageLib} from "./lib/MinimalDelegationStorageLib.sol";
 
-abstract contract MinimalDelegation is IMinimalDelegation {}
+contract MinimalDelegation {
+    using KeyLib for Key;
+
+    function authorize(Key memory key) external returns (bytes32 keyHash) {
```

Comment:
add a test? if you wanna do in followup pr thats fine but we should add unit tests as we go